diff --git a/tools/xorg-macros/Makefile b/tools/xorg-macros/Makefile
index 54df402..b36eac5 100644
--- a/tools/xorg-macros/Makefile
+++ b/tools/xorg-macros/Makefile
@@ -8,11 +8,11 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=xorg-macros
-PKG_VERSION:=1.11.0
+PKG_VERSION:=1.18.0
 
 PKG_SOURCE_URL:=http://xorg.freedesktop.org/releases/individual/util/
 PKG_SOURCE:=util-macros-$(PKG_VERSION).tar.bz2
-PKG_MD5SUM:=22d5cdff672450cb6902e0d68c200dcb
+PKG_MD5SUM:=6c4d580dc31e6b2c7be738622b9cc546
 
 HOST_BUILD_DIR:=$(BUILD_DIR_HOST)/util-macros-$(PKG_VERSION)
 
