diff --git a/utils/dbus/Makefile b/utils/dbus/Makefile
index acaac2c..1a7b274 100644
--- a/utils/dbus/Makefile
+++ b/utils/dbus/Makefile
@@ -9,22 +9,19 @@ include $(TOPDIR)/rules.mk
 
 # Make sure to also update the dbus-x package
 PKG_NAME:=dbus
-PKG_VERSION:=1.4.14
-PKG_RELEASE:=2
+PKG_VERSION:=1.7.8
+PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=http://dbus.freedesktop.org/releases/dbus/
-PKG_MD5SUM:=ae6de2562a57516cfabaf56903375ba9
+PKG_MD5SUM:=ca49eca5719f16a58b2399617ecd1ff6
 
 PKG_FIXUP:=autoreconf
 PKG_INSTALL:=1
 
+include $(INCLUDE_DIR)/host-build.mk
 include $(INCLUDE_DIR)/package.mk
 
-TARGET_LDFLAGS+= \
-	-Wl,-rpath-link=$(STAGING_DIR)/usr/lib \
-	-Wl,-rpath=/usr/lib/
-
 define Package/dbus/Default
   SECTION:=utils
   CATEGORY:=Utilities
@@ -74,12 +71,43 @@ $(call Package/dbus/Default/description)
  This package contains D-Bus utilities.
 endef
 
+HOST_CONFIGURE_VARS += \
+	ac_cv_have_abstract_sockets="yes" \
+	ac_cv_lib_expat_XML_ParserCreate_MM="yes"
+
+HOST_CONFIGURE_ARGS += \
+	--enable-shared \
+	--enable-static \
+	--disable-abstract-sockets \
+	--disable-ansi \
+	--disable-asserts \
+	--disable-console-owner-file \
+	--disable-doxygen-docs \
+	--disable-compiler_coverage \
+	--disable-selinux \
+	--disable-tests \
+	--disable-verbose-mode \
+	--disable-xml-docs \
+	--with-dbus-user=root \
+	--with-dbus-daemondir="$(STAGING_DIR_HOST)/bin" \
+	--with-system-socket="$(STAGING_DIR_HOST)/var/run/dbus/system_bus_socket" \
+	--with-system-pid-file="$(STAGING_DIR_HOST)/var/run/dbus.pid" \
+	--without-x \
+	--libexecdir="$(STAGING_DIR_HOST)/lib/dbus-1"
 
 define Build/Prepare
 	$(Build/Prepare/Default)
 	$(SED) 's/-Wl,--gc-sections/--gc-sections/' $(PKG_BUILD_DIR)/configure
 endef
 
+TARGET_LDFLAGS+= \
+	-Wl,-rpath-link=$(STAGING_DIR)/usr/lib \
+	-Wl,-rpath=/usr/lib/
+
+CONFIGURE_VARS+= \
+	ac_cv_have_abstract_sockets="yes" \
+	ac_cv_lib_expat_XML_ParserCreate_MM="yes" \
+
 CONFIGURE_ARGS += \
 	--enable-shared \
 	--enable-static \
@@ -99,12 +127,12 @@ CONFIGURE_ARGS += \
 	--with-system-socket="/var/run/dbus/system_bus_socket" \
 	--with-system-pid-file="/var/run/dbus.pid" \
 	--without-x \
-	--libexecdir=/usr/lib/dbus-1
-
-CONFIGURE_VARS+= \
-	ac_cv_have_abstract_sockets="yes" \
-	ac_cv_lib_expat_XML_ParserCreate_MM="yes" \
+	--libexecdir="/usr/lib/dbus-1"
 
+define Build/Prepare
+	$(Build/Prepare/Default)
+	$(SED) 's/-Wl,--gc-sections/--gc-sections/' $(PKG_BUILD_DIR)/configure
+endef
 
 define Build/InstallDev
 	$(INSTALL_DIR) $(1)/usr/include
@@ -182,6 +210,7 @@ define Package/dbus-utils/install
 		$(1)/usr/bin/
 endef
 
+$(eval $(call HostBuild))
 $(eval $(call BuildPackage,libdbus))
 $(eval $(call BuildPackage,dbus))
 $(eval $(call BuildPackage,dbus-utils))
