diff --git a/utils/dialog/Makefile b/utils/dialog/Makefile
index 872ab83..15489f6 100644
--- a/utils/dialog/Makefile
+++ b/utils/dialog/Makefile
@@ -8,12 +8,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=dialog
-PKG_VERSION:=1.1-20120706
+PKG_VERSION:=1.2-20140112
 PKG_RELEASE:=2
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tgz
 PKG_SOURCE_URL:=ftp://invisible-island.net/dialog
-PKG_MD5SUM:=2e538305977178eb085a9859511c299d
+PKG_MD5SUM:=373dbe256bd3bcb37f5172aa77af9fa5
 
 include $(INCLUDE_DIR)/package.mk
 
@@ -29,6 +29,8 @@ define Package/dialog/description
  A script-interpreter which provides a set of curses widgets.
 endef
 
+CONFIGURE_ARGS+= --with-ncurses
+
 define Package/dialog/install
 	$(INSTALL_DIR) $(1)/usr/bin
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/dialog $(1)/usr/bin
